sdk/index.ts

```diff
@@ -242,66 +236,69 @@ export class Flagsmith {
             Flagsmith, e.g. {"num_orders": 10}
      * @returns Segments that the given identity belongs to.
      */
-    getIdentitySegments(
+    async getIdentitySegments(
         identifier: string,
         traits?: { [key: string]: any }
     ): Promise<SegmentModel[]> {
         if (!identifier) {
             throw new Error('`identifier` argument is missing or invalid.');
         }
+        if (!this.enableLocalEvaluation) {
+            this.logger.error('This function is only permitted with local evaluation.');
+            return Promise.resolve([]);
+        }
 
         traits = traits || {};
-        if (this.enableLocalEvaluation) {
-            return new Promise((resolve, reject) => {
-                return this.environmentPromise!.then(() => {
-                    const identityModel = this.getIdentityModel(
-                        identifier,
-                        Object.keys(traits || {}).map(key => ({
-                            key,
-                            value: traits?.[key]
-                        }))
-                    );
+        const environment = await this.getEnvironment();
+        const identityModel = this.getIdentityModel(
+            environment,
+            identifier,
+            Object.keys(traits || {}).map(key => ({
+                key,
+                value: traits?.[key]
+            }))
+        );
 
-                    const segments = getIdentitySegments(this.environment, identityModel);
-                    return resolve(segments);
-                }).catch(e => reject(e));
-            });
-        }
-        console.error('This function is only permitted with local evaluation.');
-        return Promise.resolve([]);
+        return getIdentitySegments(environment, identityModel);
     }
 
-    /**
-     * Updates the environment state for local flag evaluation.
-     * Sets a local promise to prevent race conditions in getIdentityFlags / getIdentitySegments.
-     * You only need to call this if you wish to bypass environmentRefreshIntervalSeconds.
-     */
-    async updateEnvironment() {
+    private async fetchEnvironment(): Promise<EnvironmentModel> {
+        const deferred = new Deferred<EnvironmentModel>();
+        this.environmentPromise = deferred.promise;
         try {
-            const request = this.getEnvironmentFromApi();
-            if (!this.environmentPromise) {
-                this.environmentPromise = request.then(res => {
-                    this.environment = res;
-                });
-                await this.environmentPromise;
-            } else {
-                this.environment = await request;
-            }
-            if (this.environment.identityOverrides?.length) {
+            const environment = await this.getEnvironmentFromApi();
+            this.environment = environment;
```

Comment:
It's no longer clear to me when we should be setting the environment as an attribute on the class, and when we should be passing it between the methods explicitly.

Reply:
See https://github.com/Flagsmith/flagsmith-nodejs-client/pull/178#discussion_r2007660369 for more context - `this.environment` is only ever read by `getEnvironment`, and written to by `fetchEnvironment`. All methods now use `getEnvironment()` to access the environment instead of passing it around.